client/web/src/components/widgets/floating-menus/MenuList.vue

```diff
@@ -106,28 +127,38 @@ import Icon from "../labels/Icon.vue";
 import UserInputLabel from "../labels/UserInputLabel.vue";
 
 export type MenuListEntries = Array<MenuListEntry>;
+export type SectionsOfMenuListEntries = Array<MenuListEntries>;
 
-export interface MenuListEntry {
+interface MenuListEntryData {
 	label?: string;
 	icon?: string;
 	// TODO: Add `checkbox` (which overrides any `icon`)
 	shortcut?: Array<string>;
 	action?: Function;
-	children?: Array<Array<MenuListEntry>>;
-	ref?: typeof FloatingMenu | typeof MenuList;
+	children?: SectionsOfMenuListEntries;
 }
+
+export type MenuListEntry = MenuListEntryData & { ref?: typeof FloatingMenu | typeof MenuList };
+
 const MenuList = defineComponent({
 	props: {
-		direction: { type: String as PropType<MenuDirection>, value: MenuDirection.Bottom },
-		menuEntries: { type: Array as PropType<MenuListEntries>, required: true },
+		direction: { type: String as PropType<MenuDirection>, default: MenuDirection.Bottom },
+		menuEntries: { type: Array as PropType<SectionsOfMenuListEntries>, required: true },
+		activeEntry: { type: Object as PropType<MenuListEntry>, required: false },
+		minWidth: { type: Number, default: 0 },
+		defaultAction: { type: Function, required: false },
+		widthChanged: { type: Function, required: false },
```

Comment:
Prefer the use of event instead of passing functions example :

```ts
<template>
<div class="Parent">
  Parent here.
  <Child @width-change="onWidthChange" @default-action="onDefaultAction"></Child>
</div>
</template>
<script>
const Child = {
  template: "<div>Child here.</div>"
  mounted() {
    this.$emit("width-change", someValue);
    this.$emit("default-action", someValue);
  }
};

const Parent = {
  components: {
  },
  methods: {
    onWidthChange() {
      console.log("onWidthChange");
    },
    onDefaultAction() {
      console.log("onDefaultAction");
    },
  }
}
</script>
```